csrc/device_lower/pass/index.cpp

```diff
@@ -1506,7 +1507,8 @@ void IndexLowering::handle(const MmaOp* mma) {
         roundUpToMultiple(getN(mma->macro()) / 8L,
                           getBytesFromSwizzle(swizzle) / 16L) *
         /*bytes per item*/ 2L;
-    if (swizzle != MmaInputSmemSwizzle::None) {
+    if (swizzle != MmaInputSmemSwizzle::None &&
+        (mma->layout() == MmaLayout::TT || mma->layout() == MmaLayout::TN)) {
```

Comment:
Do you have some theory why this change is needed? To be honest, I'm lost here.

We handle B tensor here, but it looks like A tensor layout impacts the processing 🤔

Reply:
Yes, this is super strange. I don't have a theory :(